### DISCUSSION TITLE
Shaping points full control

### DISCUSSION THREAD

**Discussion by 0709wiwiwi** [devemux86/cruiser#589] [Q&A] [answered]
**Shaping Points full control.**

Reference: https://github.com/devemux86/cruiser/discussions/543#discussion-6697869

- Add **Shaping trackpoints** (single or fill with ) without a recalculation is offered in the Cruiser app.
- Remove **Shaping trackpoints** (single or remove all) without triggering a recalculation is not offered. 
 _For this last simple but very useful task an edit by Notepad++ is still necessary._ :-((

Use case: https://github.com/devemux86/cruiser/discussions/543#discussioncomment-9910920

**Comment by devemux86** [comment, marked as answer]
@0709wiwiwi Thanks for the suggestion.

There is already a thread about this, where the need of shaping points is also explained:
- https://github.com/devemux86/cruiser/discussions/543

**Comment by abdullahO2** [comment]
Sorry for this question that may be obvious to you, but what are the differences between Via Points and Shaping Points?

**Reply by devemux86** [reply]
- Via points are used for the stops we need to make on a journey.
They are also displayed and announced in navigation.

- Shaping points are only useful for shaping the route, i.e. making it pass through specific locations.
They are not announced in navigation.

**Reply by abdullahO2** [reply]
Thanks for the explanation!  This helps clarify things.  Let me see if I understand correctly with a couple of examples:

**Example 1 (Via Point):**

Let's say I'm driving a long distance and I know I'll need to stop for a restroom break and to grab a coffee.  I'd add the location of a rest area along my route as a *via point*.  Cruiser would then not only route me through that rest area, but it would also treat it as a destination in the navigation, telling me when I've arrived and pausing navigation until I continue.

**Example 2 (Shaping Point):**

I regularly commute on a route that has a section prone to traffic congestion.  There's a slightly longer, but usually clearer, alternative route I could take that bypasses this congestion.  I wouldn't want to stop at any particular location on this alternative route, I just want Cruiser to guide me along it.  I would add a location along this alternative route as a *shaping point*.  Cruiser would then calculate the route to pass through that point, effectively guiding me along the desired alternative route, without treating it as a navigation stop.

**Reply by devemux86** [reply]
@abdullahO2 Your examples are correct! :+1:

**Reply by 0709wiwiwi** [reply]
@abdullahO2 
See also the informative wiki post (strict vs non strict) by MotoUkRider.
https://github.com/devemux86/cruiser/wiki/Route-planning-guide

**Reply by abdullahO2** [reply]
@devemux86 @0709wiwiwi Thank you both for the clarification and the link to the wiki! The explanations and the distinction between strict and non-strict routing are very helpful.  I appreciate the additional information.
